tests/unit/test.config.js

```diff
@@ -456,14 +487,232 @@ describe('config', () => {
+        commandOpt: {
+          'api-url': {
+            requiresArg: true,
+            type: 'string',
+            demand: false,
+            default: 'pretend-default-value-of-apiKey',
+          },
+        },
+      });
+      const configObject = {
+        sign: {
+          'api-url': 2,
+        },
+      };
+      assert.throws(() => {
+        applyConf({...params, configObject});
+      }, UsageError, 'UsageError: The config option "api-url"' +
+        ' must be specified in camel case: "apiUrl"');
+    });
+
+    it('throws an error when the option is invalid', () => {
+      const params = makeArgv({
+        userCmd: ['sign'],
+        command: 'sign',
+        commandOpt: {
+          'api-url': {
+            requiresArg: true,
+            type: 'string',
+            demand: false,
+            default: 'pretend-default-value-of-apiKey',
+          },
+        },
+      });
+      const configObject = {
+        sign: {
+          randomOption: 'random-value',
+        },
+      };
+      assert.throws(() => {
+        applyConf({...params, configObject});
+      }, UsageError, 'UsageError: The config file at ' +
+      'some/path/to/config.js specified an unknown option: "randomOption"');
+    });
+
+    it('throws an error when the type of option value is invalid', () => {
+      const params = makeArgv({
+        userCmd: ['sign'],
+        command: 'sign',
+        commandOpt: {
+          'api-url': {
+            requiresArg: true,
+            type: 'string',
+            demand: false,
+            default: 'pretend-default-value-of-apiKey',
+          },
+        },
+      });
+      const configObject = {
+        sign: {
+          apiUrl: 2,
+        },
+      };
+      assert.throws(() => {
+        applyConf({...params, configObject});
+      }, UsageError, 'UsageError: The config file at some/path/to/config.js ' +
+        'specified the type of "apiUrl" incorrectly');
+    });
+
+    it('throws an error when the type of one of option values' +
+      ' is invalid', () => {
+      const params = makeArgv({
+        userCmd: ['sign'],
+        command: 'sign',
+        commandOpt: {
+          'api-url': {
+            requiresArg: true,
+            type: 'string',
+            demand: false,
+            default: 'pretend-default-value-of-apiKey',
+          },
+          'api-key': {
+            requiresArg: true,
+            type: 'string',
+            demand: false,
+            default: 'pretend-default-value-of-apiKey',
+          },
+        },
+      });
+      const configObject = {
+        sign: {
+          apiUrl: 2,
+          apiKey: 'fake-api-key',
+        },
+      };
+      assert.throws(() => {
+        applyConf({...params, configObject});
+      }, UsageError, 'UsageError: The config file at some/path/to/config.js ' +
+        'specified the type of "apiUrl" incorrectly');
+    });
+
+    it('throws an error when the type of option in config is missing', () => {
```

Comment:
This test spec shouldn't use the word config because that's misleading. This test is simulating a case where someone had programmed web-ext itself incorrectly (with a missing option type).